pyface/data_view/abstract_value_type.py

```diff
@@ -0,0 +1,180 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Provides an AbstractValueType ABC for Pyface data models.
+
+This module provides an ABC for data view value types, which are responsible
+for adapting raw data values as used by the data model's ``get_value`` and
+``set_value`` methods to the data channels that the data view expects, such
+as text, color, icons, etc.
+
+It is up to the data view to take this standardized data and determine what
+and how to actually display it.
+"""
+
+from traits.api import ABCHasStrictTraits, Event, Str, observe
+
+
+class AbstractValueType(ABCHasStrictTraits):
+    """ A value type converts raw data into data channels.
+
+    The data channels are editor value, text, color, image, and description.
+    The data channels are used by other parts of the code to produce the actual
+    display.
+
+    Subclasses should mark traits that potentially affect the display of values
+    with ``update=True`` metdadata, or alternativelym fire the ``updated``
+    event when the state of the value type changes.
+    """
+
+    #: Fired when a change occurs that requires updating values.
+    updated = Event
+
+    def can_edit(self, model, row, column):
+        """ Return whether or not the value can be edited.
+
+        The default implementation is that cells that can be set are
+        editable.
+
+        Parameters
+        ----------
+        model : AbstractDataModel
+            The data model holding the data.
+        row : sequence of int
+            The row in the data model being queried.
+        column : sequence of int
+            The column in the data model being queried.
+
+        Returns
+        -------
+        can_edit : bool
+            Whether or not the value is editable.
+        """
+        return model.can_set_value(row, column)
+
+    def get_editable(self, model, row, column):
```

Comment:
The editable value is given to the underlying toolkit to set the initial value of the editor, at least on Qt (currently Wx only supports text editing for now; eventually a custom renderer will need to be written to handle more generic objects).  The result of editing value is then passed back via `set_editable`, which is how changes get made.

I noticed a copy-paste in `set_editable`s docstring which probably isn't helping.

Reply:
`set_editable`s new docstring definitely helps but I'm not sure it completely removes the ambiguity of what "editable" means in this context. It's hard for me to judge because now I know what it means. Another pair of eyes on this might be good.

Reply:
Would a name-change for the method help?  Eg. `get_editable_value` or `get_edit_value`?

Reply:
I can't think of a better name for the method, expanding the docstring would probably bring more value. Maybe simply adding "Return a value suitable for editing by the toolkit." or something similar would make it clear.

Reply:
I have renamed to `*_editor_value` which I think is reasonably descriptive.  The only potenital confusion is over which way the get and set go, but it should hopefully be clear by analogy.